### PR TITLE
Refactor(e2e): Run Android tests on Azure Devops emulators instead of AppCenter

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Consolidating all the Android runners that used to be divided into ___DeviceAndroidRunner and ___ModuleAndroidRunner into just ___AndroidRunner.

In working on this PR, I found out that file upload tests were unintentionally being skipped for android. They do seem to fail when run, albeit it appears to be a bug in the test logic, not the actual file upload code. I'll open a separate PR to fix that.

I'm still fiddling with the test groupings to lower the build time, but it takes roughly 35 minutes for all the android tests to finish

### CODE REVIEW

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/setup/iothub/DeviceTwinCommon.java [hunk 1]
```diff
@@ -48,9 +48,9 @@
 {
     // Max time to wait to see it on Hub
     protected static final long PERIODIC_WAIT_TIME_FOR_VERIFICATION = 1000; // 1 sec
-    protected static final long MAX_WAIT_TIME_FOR_VERIFICATION = 180000; // 180 sec
```
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/setup/iothub/DeviceTwinCommon.java:51]: This timeout was simply too long. Any test timeout failures will be caught in the first 30 seconds

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java [hunk 2]
```diff
@@ -5,28 +5,13 @@
...
-import org.junit.After;
-import org.junit.Rule;
 import org.junit.Test;
 
-// This class is to add a dummy test for every TestGroup.  Currently, App Center errors out
-// when a TestGroup did not execute any actual tests.
+// This class is to add a dummy test for every TestGroup.  Currently, android testing errors out
+// when a TestGroup did not execute any actual tests, so this forces each test group to have at least
+// this test
 public class DummyAndroidRunner
```
Comment (abhipsaMisra) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java:13]: Is this really a "dummy"? Else we might want to rename it to something else

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java [hunk 3]
```diff
@@ -44,30 +29,8 @@ public void labelSnapshot()
     @TestGroup15
     @TestGroup16
     @TestGroup17
```
Comment (abhipsaMisra) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java:31]: q - we need fewer test groups now because...? can we parallelize more stuff in the android emulator, or do we spin up more emulators?

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java [hunk 4]
```diff
@@ -5,28 +5,13 @@
...
-import org.junit.Rule;
 import org.junit.Test;
 
-// This class is to add a dummy test for every TestGroup.  Currently, App Center errors out
-// when a TestGroup did not execute any actual tests.
+// This class is to add a dummy test for every TestGroup.  Currently, android testing errors out
+// when a TestGroup did not execute any actual tests, so this forces each test group to have at least
+// this test
 public class DummyAndroidRunner
 {
```
Comment (abhipsaMisra) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java:14]: Does `TestSuite` annotation work better with emulators? That way we wouldn't need a fakeTest class and a lot of other overhead code.

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java (hunk 3 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java:31]: We used to have separate test groups for device and module tests. For instance, we had a SendMessagesDeviceAndroidRunner and a SendMessagesModuleAndroidRunner, but now those two are combined into just SendMessagesAndroidRunner. On top of that, several test groups contained only a few tests, so they were also grouped together instead of given their own test group

Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java:31]: RegistryManager tests and Service Client tests, for instance, don't need their own test group anymore. They can share

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java (hunk 2 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java:13]: Sure, I can rename this to "Fake"

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java (hunk 4 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java:14]: We do still need this. We added this fake test class for App Center, but they share the same problem as running the test on an emulator. Anytime you invoke instrumentation tests on an android device, emulator or real, at least one test needs to run. Otherwise it will report an error

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java (hunk 2 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java:13]: It exists only so that, if all other tests get filtered, at least one test runs. Android instrumentation testing fails if no tests run, counterintuitively

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/FileUploadAndroidRunner.java [hunk 5]
```diff
@@ -46,17 +44,4 @@ public static Collection inputsCommons() throws Exception
         isBasicTierHub = Boolean.parseBoolean(BuildConfig.IsBasicTierHub);
         return FileUploadTests.inputs();
     }
-
-    @Ignore
-    @Override
-    @Test
-    public void uploadToBlobAsyncSingleFileZeroLength()
```
Comment (abhipsaMisra) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/FileUploadAndroidRunner.java:53]: Q - why was this here?

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/HubTierConnectionAndroidRunner.java [hunk 6]
```diff
@@ -51,16 +44,4 @@ public static Collection inputs() throws IOException, IotHubException, GeneralSe
         identities = getIdentities(inputs);
         return inputs;
     }
-
-    @AfterClass
-    public static void cleanUpResources()
-    {
-        tearDown(identities);
```
Comment (abhipsaMisra) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/HubTierConnectionAndroidRunner.java:58]: I assume this cleanup is happening elsewhere now?

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/errorinjection/twin/ReportedPropertiesErrInjAndroidRunner.java [hunk 7]
```diff
@@ -24,17 +21,14 @@
...
+public class ReportedPropertiesErrInjAndroidRunner extends ReportedPropertiesErrInjTests
 {
     @Rule
     public Rerun count = new Rerun(3);
 
-    @Rule
-    public ReportHelper reportHelper = Factory.getReportHelper();
-
-    public ReportedPropertiesDeviceAndroidRunner(IotHubClientProtocol protocol, AuthenticationType authenticationType, ClientType clientType, String publicKeyCert, String privateKey, String x509Thumbprint)
+    public ReportedPropertiesErrInjAndroidRunner(IotHubClientProtocol protocol, AuthenticationType authenticationType, ClientType clientType, String publicKeyCert, String privateKey, String x509Thumbprint)
```
Comment (abhipsaMisra) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/errorinjection/twin/ReportedPropertiesErrInjAndroidRunner.java:31]: Q - don't we have tests that take in the SSL context directly, or do they have a specific test class?

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/messaging/SendMessagesAndroidRunner.java [hunk 8]
```diff
@@ -26,19 +23,15 @@
...
+public class SendMessagesAndroidRunner extends SendMessagesTests
 {
     @Rule
     public Rerun count = new Rerun(3);
 
-    @Rule
-    public ReportHelper reportHelper = Factory.getReportHelper();
-
-    public SendMessagesModuleAndroidRunner(IotHubClientProtocol protocol, AuthenticationType authenticationType, ClientType clientType, String publicKeyCert, String privateKey, String x509Thumbprint, boolean useHttpProxy) throws Exception
+    public SendMessagesAndroidRunner(IotHubClientProtocol protocol, AuthenticationType authenticationType, ClientType clientType, String publicKeyCert, String privateKey, String x509Thumbprint, boolean useHttpProxy) throws Exception
```
Comment (abhipsaMisra) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/messaging/SendMessagesAndroidRunner.java:34]: nit: It would help if these constructors and parameterized test methods have some comments that describe what they do. I know this has been a point of confusion for me in the past, because these tests have pretty deep inheritance, and the test class and method names are not descriptive of what they are responsible for.

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/FileUploadAndroidRunner.java (hunk 5 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/FileUploadAndroidRunner.java:53]: This test was failing on android in particular at some point, but it was never re-enabled

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/setup/iothub/DeviceTwinCommon.java (hunk 1 above)
Comment (abhipsaMisra) [reply iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/setup/iothub/DeviceTwinCommon.java:51]: Do we have any long running tests? Token refresh tests? Connection status change tests?

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/errorinjection/twin/ReportedPropertiesErrInjAndroidRunner.java (hunk 7 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/errorinjection/twin/ReportedPropertiesErrInjAndroidRunner.java:31]: We do have a single test for that in the SendMessages test suite

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/HubTierConnectionAndroidRunner.java (hunk 6 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/HubTierConnectionAndroidRunner.java:58]: Actually, this looks like it shouldn't have been deleted. I'll revert this

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/telemetry/SendMessagesTests.java [hunk 9]
```diff
@@ -58,12 +56,6 @@ public void sendSecurityMessages() throws Exception
     @Test
     public void sendMessages() throws Exception
     {
-        if (testInstance.protocol == MQTT_WS && (testInstance.authenticationType == SELF_SIGNED || testInstance.authenticationType == CERTIFICATE_AUTHORITY))
```
Comment (abhipsaMisra) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/telemetry/SendMessagesTests.java:61]: removed, why?

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/messaging/SendMessagesAndroidRunner.java (hunk 8 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/messaging/SendMessagesAndroidRunner.java:34]: Sure thing, I can add some comments. I plan on removing the JVMRunner classes and moving their logic into the test classes themselves to help cut down on the inheritance.

File: vsts/RunTestsOnEmulator.sh [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+#List instrumentation classes, for logging purposes only
+echo 'Listing available instrumentations:'
+adb -e shell pm list instrumentation
+echo ''
+
+annotationString="com.microsoft.azure.sdk.iot.android.helper.${TEST_GROUP_ID}"
+echo 'Running android tests with annotation'
+echo $annotationString
+
+#Return code from adb shell isn't returned as one would expect. Need to capture output logs and analyze them to determine if this test run was a success or not
```
Comment (abhipsaMisra) [inline vsts/RunTestsOnEmulator.sh:17]: whoa!

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/telemetry/SendMessagesTests.java (hunk 9 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/telemetry/SendMessagesTests.java:61]: MQTT_WS does support x509 auth now. It's orthogonal to this PR, but worth enabling

File: vsts/RunTestsOnEmulator.sh (hunk 10 above)
Comment (timtay-microsoft) [reply vsts/RunTestsOnEmulator.sh:17]: Yeah, I'm not happy about this. I don't have another way though

File: vsts/RunTestsOnEmulator.sh [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+echo "$TestLogs"
+
+#This string is always present in the output logs from android testing if a test failed. Hacky, but there is no return code
+# to check instead.
+FailureMessage='FAILURES!!!'
+
+if grep -q "$FailureMessage" <<< "$TestLogs"; then
+    echo 'Test failures detected, exiting...'
+    exit -1
+fi
```
Comment (abhipsaMisra) [inline vsts/RunTestsOnEmulator.sh:30]: Do we get test results on the AzureDevOps pane, or will we have to parse the output log file?

File: vsts/StartEmulator.sh [hunk 12]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+echo 'Listing available android sdks for installation'
+$ANDROID_HOME/tools/bin/sdkmanager --list | grep system-images
+
+emulatorImage='system-images;android-28;google_apis;x86_64'
```
Comment (abhipsaMisra) [inline vsts/StartEmulator.sh:6]: are we testing just a single API version, or the mix and max that we claim to support?

File: vsts/RunTestsOnEmulator.sh (hunk 11 above)
Comment (timtay-microsoft) [reply vsts/RunTestsOnEmulator.sh:30]: For now, you only get the logs. I'm still investigating how to improve this experience

File: vsts/StartEmulator.sh (hunk 12 above)
Comment (timtay-microsoft) [reply vsts/StartEmulator.sh:6]: We only ever tested API 28 on App center, and this PR doesn't change that. In theory though, we could run the same tests on two or more android API versions

Comment (timtay-microsoft) [reply vsts/StartEmulator.sh:6]: The list operation on line 4 actually prints out all the alternatives to the ```system-images;android-28;google_apis;x86_64``` that can be installed

File: vsts/install_appcenter_cli.ps1 [hunk 13]
```diff
@@ -1 +0,0 @@
-﻿npm install appcenter-cli
```
Comment (abhipsaMisra) [inline vsts/install_appcenter_cli.ps1:1]: does this file still exist?

File: vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml [hunk 14]
```diff
@@ -1,299 +1,249 @@
...
+          IOTHUB_CONNECTION_STRING: $(WINDOWS-IOTHUB-CONNECTION-STRING)
+          STORAGE_ACCOUNT_CONNECTION_STRING: $(STORAGE-ACCOUNT-CONNECTION-STRING)
+          IOTHUB_DEVICE_CONN_STRING_INVALIDCERT: $(IOTHUB-DEVICE-CONN-STRING-INVALIDCERT)
+          IOTHUB_CONN_STRING_INVALIDCERT: $(IOTHUB-CONN-STRING-INVALIDCERT)
+          DPS_GLOBALDEVICEENDPOINT_INVALIDCERT: $(DPS-GLOBALDEVICEENDPOINT-INVALIDCERT)
+          PROVISIONING_CONNECTION_STRING_INVALIDCERT: $(PROVISIONING-CONNECTION-STRING-INVALIDCERT)
+          CUSTOM_ALLOCATION_POLICY_WEBHOOK: $(CUSTOM-ALLOCATION-POLICY-WEBHOOK)
+          FAR_AWAY_IOTHUB_CONNECTION_STRING: $(FAR-AWAY-IOTHUB-CONNECTION-STRING)
+          IS_BASIC_TIER_HUB: $(IS-BASIC-TIER-HUB)
+          TARGET_BRANCH: $(System.PullRequest.TargetBranch)
```
Comment (abhipsaMisra) [inline vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:38]: so few env var required here!

File: vsts/install_appcenter_cli.ps1 (hunk 13 above)
Comment (timtay-microsoft) [reply vsts/install_appcenter_cli.ps1:1]: No, that file is deleted

File: vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml [hunk 15]
```diff
@@ -1,299 +1,249 @@
...
+          targetType: 'filePath'
+          filePath: '$(Build.SourcesDirectory)/vsts/StartEmulator.sh'
+
+      - task: Bash@3
+        displayName: 'Run tests on emulator'
+        inputs:
+          targetType: 'filePath'
+          filePath: '$(Build.SourcesDirectory)/vsts/RunTestsOnEmulator.sh'
+        env:
+          TEST_GROUP_ID: $(ANDROID_TEST_GROUP_ID)
```
Comment (timtay-microsoft) [inline vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:249]: Note that the android environment variable values are set during build time, not runtime. The APKs that are generated from a build contain the environment variables such as iothub connection string. As such, there is no need to include them here, too

**Comment by timtay-microsoft** [comment]
> I would like to sync up with you to see how these emulators get spawned.

Here's some good reading, then: https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/android?view=azure-devops

StartEmulator.sh was largely based on my learnings from that document.

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/errorinjection/twin/ReportedPropertiesErrInjAndroidRunner.java (hunk 7 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/errorinjection/twin/ReportedPropertiesErrInjAndroidRunner.java:31]: For context, the SSLContext constructor was a very recent addition, and most of these tests were written before then.

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/setup/iothub/DeviceTwinCommon.java (hunk 1 above)
Comment (abhipsaMisra) [reply iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/setup/iothub/DeviceTwinCommon.java:51]: ping on this question

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java (hunk 4 above)
Comment (abhipsaMisra) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java]: Sure, but with `TestSuite`, instead of multiple runners, we can have a single runner that references all the test files, and does the initial setup.

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/FileUploadAndroidRunner.java (hunk 5 above)
Comment (abhipsaMisra) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/FileUploadAndroidRunner.java:53]: are we keeping this now?

File: vsts/RunTestsOnEmulator.sh (hunk 11 above)
Comment (abhipsaMisra) [reply vsts/RunTestsOnEmulator.sh:30]: Another question then - do we get a summary at the end?

File: vsts/StartEmulator.sh (hunk 12 above)
Comment (abhipsaMisra) [reply vsts/StartEmulator.sh:6]: > In theory though, we could run the same tests on two or more android API versions

Is the effort small enough to add to this PR directly?

File: vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml [hunk 16]
```diff
@@ -1,299 +1,248 @@
...
+        inputs:
+          mergeTestResults: true
+        continueOnError: true
+        condition: always()
+
+  ### Android, Multi configuration build (12 different test groups to cover) ###
+  - job: AndroidBuild
+    timeoutInMinutes: 45
+    pool:
+      name: Hosted VS2017
```
Comment (abhipsaMisra) [inline vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:134]: The [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/android?view=azure-devops#test-on-the-android-emulator) mentioned that: The Android Emulator is currently available only on the Hosted macOS agent.

But we seem to be using a win image here?

Comment (abhipsaMisra) [reply vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:134]: Ah, I see the AndroidTest uses macOS-latest

**Comment by abhipsaMisra** [comment]
> StartEmulator.sh was largely based on my learnings from that document.

That helped, thanks. My question was how we spin up multiple emulators, and [this line](https://github.com/Azure/azure-iot-sdk-java/pull/809/files#diff-9824f3d0672b6a72627425cbefd407c0R174) answers my question.

File: vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml (hunk 16 above)
Comment (timtay-microsoft) [reply vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:134]: We build the android APKs on a windows image, but the Android Test portion of this yaml is all on a Mac OS

File: vsts/RunTestsOnEmulator.sh (hunk 11 above)
Comment (timtay-microsoft) [reply vsts/RunTestsOnEmulator.sh:30]: Yes, there is a summary logged of all the failures that happened.

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/FileUploadAndroidRunner.java (hunk 5 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/FileUploadAndroidRunner.java:53]: File upload as a whole wasn't being tested on Android, but not on purpose. I'd re-enable it but there are some test failures on Android. They appear to be test bugs, not bugs in file upload itself. I'll open a separate PR to fix that bug, and to re-enable file upload tests on android

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java (hunk 4 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/helper/DummyAndroidRunner.java]: I don't see how we could split the tests into groups with this approach

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/messaging/SendMessagesAndroidRunner.java (hunk 8 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/iothub/messaging/SendMessagesAndroidRunner.java:34]: This effort will be in a different PR, since removing the JVMRunner classes is a big task

File: vsts/StartEmulator.sh (hunk 12 above)
Comment (timtay-microsoft) [reply vsts/StartEmulator.sh:6]: Sure, but we haven't had a requirement like that yet. I could see that being a separate PR, if we need it at all
